pkg/reboot/reboot.go

```diff
@@ -0,0 +1,109 @@
+// Package reboot provides a function to reboot the system.
+package reboot
+
+import (
+	"bufio"
+	"context"
+	"errors"
+	"fmt"
+	stdos "os"
+	"strings"
+
+	"github.com/leptonai/gpud/log"
+	"github.com/leptonai/gpud/pkg/process"
+)
+
+type Op struct {
+	useSystemctl             bool
+	fileToCreateBeforeReboot string
+}
+
+type OpOption func(*Op)
+
+func (op *Op) applyOpts(opts []OpOption) error {
+	for _, opt := range opts {
+		opt(op)
+	}
+
+	return nil
+}
+
+// Set true to run "systemctl reboot".
+func WithSystemctl(b bool) OpOption {
+	return func(op *Op) {
+		op.useSystemctl = b
+	}
+}
+
+// Specifies a file to create before rebooting.
+func WithFileToCreateBeforeReboot(file string) OpOption {
```

Comment:
these two functions (With*) won't be needed, we can just do a generic reboot, custom file injection will be done on node initialization.

Reply:
removed